api/contrib/envoy/extensions/matching/input_matchers/hyperscan/v3alpha/hyperscan.proto

```diff
@@ -0,0 +1,83 @@
+syntax = "proto3";
+
+package envoy.extensions.matching.input_matchers.hyperscan.v3alpha;
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.matching.input_matchers.hyperscan.v3alpha";
+option java_outer_classname = "HyperscanProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/matching/input_matchers/hyperscan/v3alpha";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Hyperscan matcher]
+// [#extension: envoy.matching.input_matchers.hyperscan]
+
+// `Hyperscan <https://github.com/intel/hyperscan>`_ regex matcher. The engine of the matcher uses
+// hybrid automata techniques to allow simultaneous matching of large numbers of regular
+// expressions across streams of data.
+//
+// The matcher follows PCRE pattern syntax, and the regex string must adhere to the documented
+// `pattern support <https://intel.github.io/hyperscan/dev-reference/compilation.html#pattern-support>`_.
+message Hyperscan {
+  // [#next-free-field: 14]
+  message Regex {
+    reserved 6, 10, 11;
```

Comment:
These 3 fields are reserved for Hyperscan single match (only report the first match), prefilter (compile an approximate version for prefiltering) and leftmost (return the leftmost match) ability. Since Hyperscan matcher is used as a input matcher currently, we will not use these features.

Reply:
You might want to doc that. I think IIRC this protobuf needs to be wire-compatible with another protobuf in a different codebase and that seems worth calling out in comments.

Reply:
Sure, I will document them.

Reply:
Unfortunately, the reserved fields cannot have comments on them. It is fine to remove these reserved fields directly.